mitodl/open-discussions
# Added mail app supporting sending of emails (#449)

#### What are the relevant tickets?
Fixes #449 

#### What's this PR do?
Adds the `djang-anymail` plugin for mailgun and a minimal API to render and send email messages to users who have opted in.

#### How should this be manually tested?

- Ensure you have valid settings for the following keys in .env: `MAILGUN_SENDER_DOMAIN`, `MAILGUN_KEY`, `MAILGUN_RECIPIENT_OVERRIDE`, `MAILGUN_FROM_EMAIL` (message me)
- Run this code without error to render the email and send to your `MAILGUN_RECIPIENT_OVERRIDE` address:

```python
from mail.api import *
from django.contrib.auth.models import User

recipients = User.objects.all()[:10]

messages = messages_for_recipients([
    (recipient, user, {
        'url': 'http://google.com',
    }) for recipient, user in safe_format_recipients(recipients)
], 'sample')

messages = list(messages)

assert len(messages) == 1

send_messages(messages)
```
